packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -257,6 +325,144 @@ function toCollectorInstrumentationLibrarySpans(
   };
 }
 
+/**
+ * Given a MetricDescriptor, return its type in a compatible format with the collector
+ * @param descriptor
+ */
+export function toCollectorType(
+  descriptor: MetricDescriptor
+): opentelemetryProto.metrics.v1.MetricDescriptorType {
+  if (
+    descriptor.metricKind === MetricKind.UP_DOWN_COUNTER ||
+    descriptor.metricKind === MetricKind.OBSERVER
+  ) {
+    if (descriptor.valueType === apiValueType.INT) {
+      return opentelemetryProto.metrics.v1.MetricDescriptorType.INT64;
+    }
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.DOUBLE;
+  }
+  // Monotonic counter
+  else if (descriptor.metricKind === MetricKind.COUNTER) {
+    if (descriptor.valueType === apiValueType.INT) {
+      return opentelemetryProto.metrics.v1.MetricDescriptorType.MONOTONIC_INT64;
+    }
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.MONOTONIC_DOUBLE;
+  } else {
+    // Other types not yet implemented
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.INVALID_TYPE;
+  }
+}
+
+/**
+ * Given a MetricDescriptor, return its temporality in a compatible format with the collector
+ * @param descriptor
+ */
+export function toCollectorTemporality(
+  descriptor: MetricDescriptor
+): opentelemetryProto.metrics.v1.MetricDescriptorTemporality {
+  if (descriptor.metricKind === MetricKind.COUNTER) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.CUMULATIVE;
+  } else if (descriptor.metricKind === MetricKind.OBSERVER) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality
+      .INSTANTANEOUS;
+  } else if (descriptor.metricKind === MetricKind.VALUE_RECORDER) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.DELTA;
+  }
+  return opentelemetryProto.metrics.v1.MetricDescriptorTemporality
+    .INVALID_TEMPORALITY;
+}
```

Comment:
I wasn't really sure what to do here, especially regarding Temporality. None of the other exporters have incorporated temporality yet.